packages/contracts-bedrock/scripts/deploy/Deploy.s.sol

```diff
@@ -1578,49 +1648,64 @@ contract Deploy is Deployer {
         }
 
         uint32 rawGameType = GameType.unwrap(_params.gameType);
+        FaultDisputeGameParams memory _params_ = _params;
         if (rawGameType != GameTypes.PERMISSIONED_CANNON.raw()) {
             _factory.setImplementation(
-                _params.gameType,
+                _params_.gameType,
                 IDisputeGame(
-                    _deploy(
-                        "FaultDisputeGame",
-                        string.concat("FaultDisputeGame_", vm.toString(rawGameType)),
-                        abi.encode(
-                            _params.gameType,
-                            _params.absolutePrestate,
-                            _params.maxGameDepth,
-                            cfg.faultGameSplitDepth(),
-                            cfg.faultGameClockExtension(),
-                            _params.maxClockDuration,
-                            _params.faultVm,
-                            _params.weth,
-                            IAnchorStateRegistry(mustGetAddress("AnchorStateRegistryProxy")),
-                            cfg.l2ChainID()
-                        )
-                    )
+                    DeployUtils.create2AndSave({
+                        _save: this,
+                        _name: "FaultDisputeGame",
+                        _nick: string.concat("FaultDisputeGame_", vm.toString(rawGameType)),
+                        _args: DeployUtils.encodeConstructor(
+                            abi.encodeCall(
+                                IFaultDisputeGame.__constructor__,
+                                (
+                                    _params_.gameType,
+                                    _params_.absolutePrestate,
+                                    _params_.maxGameDepth,
+                                    cfg.faultGameSplitDepth(),
+                                    _unsafeCastUint256ToDuartion(cfg.faultGameClockExtension()),
```

Comment:
`Duration.wrap(uint64(cfg.faultGameClockExtension()))`

Can use this instead

Reply:
yeah definitely